src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.ts

```diff
@@ -10,14 +14,49 @@ import { StudentExamService } from 'app/exam/manage/student-exams/student-exam.s
 export class StudentExamDetailComponent implements OnInit {
     courseId: number;
     studentExam: StudentExam;
+    course: Course;
+    student: User;
 
-    constructor(private route: ActivatedRoute, private studentExamService: StudentExamService) {}
+    constructor(private route: ActivatedRoute, private studentExamService: StudentExamService, private courseService: CourseManagementService) {}
 
     /**
      * Initialize the courseId and studentExam
      */
     ngOnInit(): void {
+        this.loadAll();
+    }
+
+    loadAll() {
         this.courseId = Number(this.route.snapshot.paramMap.get('courseId'));
         this.route.data.subscribe(({ studentExam }) => (this.studentExam = studentExam));
+
+        this.courseService.find(this.courseId).subscribe((courseResponse) => {
+            this.course = courseResponse.body!;
+        });
+        // @ts-ignore
+        this.student = this.studentExam.user;
```

Comment:
Don't ignore useful type warning :-D 

I think this should be `this.studentExam.student` or am I mistaken?

Reply:
fixed